.github/workflows/bindings_java_snapshot.yml

```diff
@@ -0,0 +1,126 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: Deploy Java Binding Snapshot
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+    paths:
+      - "bindings/java/**"
+      - ".github/workflows/bindings_java.yml"
+  schedule:
+      # At 01:30 on Monday, every Monday.
+      - cron: '30 1 * * 1'
+  workflow_dispatch:
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  stage-snapshot:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        include:
+        - os: ubuntu-latest
+          classifier: linux-x86_64
```

Comment:
Where is the naming rule from? Can we use `windows_x86_64`?

Reply:
Let's say basically you can automate everything with scripts, and I have implemented it days before. Here are a few problems:

1. __Windows and PowerShell compatibility issues__;
2. Resolve the classifier by yourself - it effectively reimplements core logic in os-maven-plugin;
3. Pass the resolved classifier to the Java environment.

I _can_ do it, of course. But if we can just live a normal life, I don't want to actually do it.

Reply:
The tricky part is that you can hardly reuse os-maven-plugin beyond the Maven plugin lifecycle.